inventory/src/components/KitIndex/KitAdd.js

```diff
@@ -0,0 +1,236 @@
+import AddIcon from '@mui/icons-material/Add';
+import CancelIcon from '@mui/icons-material/Cancel';
+import SendIcon from '@mui/icons-material/Send';
+import { LoadingButton } from '@mui/lab';
+import {
+  Box,
+  FormControl,
+  IconButton,
+  InputLabel,
+  MenuItem,
+  Modal,
+  Select,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { useFormik } from 'formik';
+import React, { useEffect, useState } from 'react';
+
+import { useCreateKit } from '../../hooks/mutations/useCreateKit';
+import { useKitRecipe } from '../../hooks/queries/useKitRecipe';
+
+import { BlueprintSelect } from './BlueprintSelect';
+import { kitAddValidation } from './schema';
+
+export const KitAdd = ({ blueprints }) => {
+  const [open, setOpen] = useState(false);
+  const [selectedBlueprint, setSelectedBlueprint] = useState('');
+  const { data: kitRecipeData, isLoading: blueprintLoading } =
+    useKitRecipe(selectedBlueprint);
+  const { mutate, isLoading: createKitLoading } = useCreateKit({});
+
+  const formik = useFormik({
+    initialValues: {
+      name: '',
+      blueprint: '',
+      kitContent: {},
+      itemTotalQty: {},
+    },
+    validationSchema: kitAddValidation,
+    validateOnChange: false,
+    onSubmit: () => {
+      const content = Object.values(formik.values.kitContent)
+        .map((v) => Object.values(v))
+        .flat();
+      const payload = {
+        name: formik.values.name,
+        blueprint: formik.values.blueprint,
+        content,
+      };
+      mutate(payload, {
+        onSuccess: () => {
+          onClose();
+        },
+      });
+    },
+  });
+
+  useEffect(() => {
+    if (!kitRecipeData) return;
+    const newKitContent = {};
+    for (const item of kitRecipeData) {
+      for (const expiry of item.item_options) {
+        if (!newKitContent[item.item_id]) newKitContent[item.item_id] = {};
+        newKitContent[item.item_id][expiry.item_expiry_id] = {
+          item_expiry_id: expiry.item_expiry_id,
+          quantity: 0,
+          expiryTotalQty: expiry.quantity,
+          itemTotalQty: item.required_quantity,
+        };
+      }
+    }
+    formik.setFieldValue('kitContent', newKitContent);
+  }, [kitRecipeData, formik]);
```

Comment:
```suggestion
      // adding formik into dependency array causes state update issues in jest
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [kitRecipeData]);
```

Remove formik in the dependency array and add the eslint-disable-next-line to disable the warning. Seems like adding formik will cause state updating issues in jest. That would hopefully fix the CI issue